src/main/java/org/openrewrite/java/migrate/lang/NullCheck.java

```diff
@@ -0,0 +1,149 @@
+
+import lombok.Value;
+import org.jspecify.annotations.Nullable;
+import org.openrewrite.Cursor;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.search.SemanticallyEqual;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.Statement;
+import org.openrewrite.trait.SimpleTraitMatcher;
+import org.openrewrite.trait.Trait;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+@Value
+public class NullCheck implements Trait<J.If> {
+    Cursor cursor;
+    Expression nullCheckedParameter;
+
+    public Statement whenNull() {
+        return getTree().getThenPart();
+    }
+
+    public @Nullable Statement whenNotNull() {
+        J.If.Else anElse = getTree().getElsePart();
+        if (anElse == null) {
+            return null;
+        }
+        return anElse.getBody();
+    }
+
+    public boolean returns() {
+        Statement statement = whenNull();
+        if (statement instanceof J.Block) {
+            for (Statement s : ((J.Block) statement).getStatements()) {
+                if (s instanceof J.Return) {
+                    return true;
+                }
+            }
+            return false;
+        }
+        return statement instanceof J.Return;
+    }
+
+    /**
+     * Calculates few potential cases where the null checked variable gets reassigned and only returns false if these cases DO NOT match.
+     * In any other case this returns true as we do not know that particular situation yet -> no harm -> assume it could be altered in the block.
+     * @return false only if we are 100% sure the block does not reassigns/changes the null checked variable.
+     */
+    public boolean couldModifyNullCheckedValue() {
+        Statement statement = whenNull();
+        if (statement instanceof J.Block || statement instanceof Expression) {
+            return couldModifyNullCheckedValue(statement, nullCheckedParameter);
+        }
+        // Cautious by default
+        return true;
+    }
+    private static boolean couldModifyNullCheckedValue(J expression, Expression nullChecked) {
+        if (nullChecked instanceof J.FieldAccess && couldModifyNullCheckedValue(expression, ((J.FieldAccess) nullChecked).getTarget())) {
+            return true;
+        }
+        if (nullChecked instanceof J.MethodInvocation &&
+                ((J.MethodInvocation) nullChecked).getSelect() != null &&
+                couldModifyNullCheckedValue(expression, ((J.MethodInvocation) nullChecked).getSelect())) {
+            return true;
+        }
+        return new JavaIsoVisitor<AtomicBoolean>() {
+
+            private final boolean isCertainlyImmutable = nullChecked.getType() != null && JavaType.Primitive.fromClassName(nullChecked.getType().toString()) != null;
+
+            @Override
+            public J.Identifier visitIdentifier(J.Identifier identifier, AtomicBoolean atomicBoolean) {
```

Comment:
```suggestion
            public J.Identifier visitIdentifier(J.Identifier identifier, AtomicBoolean couldModifyNull) {
```

Given a variable the name of the type is a bad practice. Use a name that represents what it intention it represents.